File: JS/NextJS/nest-next-example/src/user/user.controller.ts
import {Controller, Get, Render} from "@nestjs/common";
import {UserService} from "./user.service";
import {User} from "./user.entity";

@Controller('users')
export class UserController {
  constructor(
    private readonly userService: UserService
  ) {}

  // view 용
  @Render('user/user-list')
  @Get('user-list')
  userListView(): void {
  }

  // REST API
  @Get()
  getUsers(): User[] {
    return this.userService.getAllUsers();
  }
}

File: java/spring-boot-gradle-multi-module/api/src/main/java/io/github/ivvve/api/user/UserController.java
package io.github.ivvve.api.user;

import io.github.ivvve.user.entity.User;
import io.github.ivvve.user.service.UserService;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;

@RequiredArgsConstructor
@RestController
@RequestMapping("users")
public class UserController {
    private final UserService userService;

    @GetMapping
    public Flux<User> getAllUsers() {
        return this.userService.getAll();
    }

    @PostMapping
    public Flux<User> addNewUser(@RequestBody final AddNewUserRequest request) {
        return this.userService.addUser(request.name, request.nickname)
                .flatMapMany(u -> this.userService.getAll());
    }
}
